internal/terraform/module/module_ops.go

```diff
@@ -359,19 +363,47 @@ func ParseModuleConfiguration(ctx context.Context, fs ReadOnlyFS, modStore *stat
 
 	// TODO: Avoid parsing if the content matches existing AST
 
-	// TODO: Only parse the file that's being changed/opened, unless this is 1st-time parsing
-
 	// Avoid parsing if it is already in progress or already known
 	if mod.ModuleParsingState != op.OpStateUnknown && !job.IgnoreState(ctx) {
 		return job.StateNotChangedErr{Dir: document.DirHandleFromPath(modPath)}
 	}
 
-	err = modStore.SetModuleParsingState(modPath, op.OpStateLoading)
+	var files ast.ModFiles
+	var diags ast.ModDiags
+	rpcContext := lsctx.RPCContext(ctx)
+	// Only parse the file that's being changed/opened, unless this is 1st-time parsing
+	if mod.ModuleParsingState == op.OpStateLoaded && rpcContext.IsDidChangeRequest() {
```

Comment:
```suggestion
	if mod.ModuleParsingState == op.OpStateLoaded && rpcContext.IsDidChangeRequest() && lsctx.IsLanguageId(ctx, ilsp.Terraform.String()) {
```
This is important so that we don't end up parsing tfvars files as if they're part of the module ("module" meaning the collection of `*.tf` and `*.tf.json` files).

On a related note, there's an opportunity for another future optimisation PR similar to this one, which involves `ParseVariables` job below.